### PR TITLE
Create types for react-google-picker

### PR DESCRIPTION
**Pull request by ebroder**
I've written up a type declaration for the [`react-google-picker` package](https://www.npmjs.com/package/react-google-picker). The tests are taken from the example code, as was the type declaration (except for when the documentation wasn't sufficiently clear, in which case I went to the source code behavior)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-google-picker/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Type definitions for react-google-picker 0.1
+// Project: https://github.com/sdoomz/react-google-picker
+// Definitions by: Evan Broder <https://github.com/me>
```
Comment (DangerBotOSS) [inline types/react-google-picker/index.d.ts:3]: - :no_entry_sign: This line should have your github username in it, not /me

File: types/react-google-picker/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Type definitions for react-google-picker 0.1
+// Project: https://github.com/sdoomz/react-google-picker
+// Definitions by: Evan Broder <https://github.com/ebroder>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.0
```
Comment (peterblazejewicz) [inline types/react-google-picker/index.d.ts:5]:
```suggestion
```
why this one was added? (minimum version)

File: types/react-google-picker/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    scope?: ReadonlyArray<string>;
+    viewId?: keyof typeof google.picker.ViewId;
+    mimeTypes?: ReadonlyArray<string>;
+    query?: string;
+    authImmediate?: boolean;
+    origin?: string;
+    onChange?: (result: google.picker.ResponseObject) => void;
+    onAuthenticate?: (oauthToken: string) => void;
+    onAuthFailed?: (response: GoogleApiOAuth2TokenObject) => void;
+    createPicker?: (google: typeof window.google, oauthToken: string) => void;
```
Comment (peterblazejewicz) [inline types/react-google-picker/index.d.ts:24]:
```suggestion
    createPicker?: (google: typeof self.google, oauthToken: string) => void;
```
IMO this is more proper:
https://developer.mozilla.org/en-US/docs/Web/API/Window/self

File: types/react-google-picker/index.d.ts (hunk 2 above)
Comment (ebroder) [reply types/react-google-picker/index.d.ts:5]: I think I was trying to figure out how to clear this warning:

```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.8.0

YOUR TYPESCRIPT VERSION: 4.9.0-dev.20220811

Please only submit bug reports when using the officially supported version.
```

but it's not necessary - I'll drop it.

**Comment by ebroder** [comment]
Sorry, I amended by habit instead of adding a new commit, but the change is just your two suggestions. Thanks for the notes.

File: types/react-google-picker/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/react-google-picker/index.d.ts]: 4.8 is RC (next), and the current tooling in ecosystem, does not support it. DT is being used by TS team as proving ground for upcoming TS versions

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ebroder thanks!

**Comment by ebroder** [comment]
Ready to merge

**Pull request merged**